src/SingleContainer/Worker/NetworkPerspective.Sync.Worker.Application/Services/ISyncService.cs

```diff
@@ -65,12 +68,20 @@ public async Task<SyncResult> SyncAsync(SyncContext context, CancellationToken s
             await SyncEntitiesAsync(_dataSource, context, stoppingToken);
             var syncResult = await SyncInteractionsAsync(_dataSource, context, stoppingToken);
 
-            await _networkPerspectiveCore.ReportSyncSuccessfulAsync(context.AccessToken, context.TimeRange, stoppingToken);
+            await _networkPerspectiveCore.ReportSyncSuccessfulAsync(context.AccessToken, context.TimeRange,
+                stoppingToken);
             await _statusLogger.LogInfoAsync("Sync completed", stoppingToken);
             _logger.LogInformation("Synchronization completed for Connector '{connectorId}'", context.ConnectorId);
 
             return syncResult;
         }
+        catch (ValidationException ve)
+        {
+            await _networkPerspectiveCore.TryReportSyncFailedAsync(context.AccessToken, context.TimeRange, ve.Message, stoppingToken);
+            _logger.LogError(ve, "Cannot complete synchronization job for Connector {connectorId}.\n{exceptionMessage}", context.ConnectorId, ve.Message);
+            await _statusLogger.LogErrorAsync($"Sync cancelled because of validation error '{ve.Message}'", CancellationToken.None);
+            return SyncResult.Empty;
+        }
```

Comment:
Where from the ValidationException might be thrown?

Reply:
Excel connector when number of records is too small. The message - fail reason - should be logged and displayed to user.